src/qemu.rs

```diff
@@ -691,7 +689,27 @@ impl Qemu {
             _init: init,
             updates,
             image: target.image.is_some(),
-        })
+        };
+
+        // We still need to possibly redirect the standard streams.
+        // By default, when calling `spawn`, the child process inherits
+        // the standard streams of the parent. This is not what we want
+        // when running in non-interactive mode.
+        if !qemu.interactive() {
+            qemu.process
+                .stdin(Stdio::null())
+                .stdout(Stdio::piped())
+                .stderr(Stdio::piped());
+        }
+        Ok(qemu)
+    }
+
+    /// Return whether or not this target is ran in interactive mode
+    ///
+    /// Interactive mode means we are not running a command and we are not
+    /// running in image mode.
+    fn interactive(&self) -> bool {
+        self.command == MAGIC_INTERACTIVE_COMMAND && !self.image
```

Comment:
@chantra why `!self.image`?

Reply:
IIRC, but I have not played with images much, the image does not depend on a init file, which is what provides the root prompt here. I don't think it is just a matter of removing the check on whether or not this is running in an image to have it magically work, but I may be wrong.